### PR TITLE
SRV-1588: Migrate flagged S3 config values to SSM

### PR DESCRIPTION
**Pull request by afaq-sf**
## Existing Behavior
- Secrets are only populated for deploy job
- SSM keys are pulled and populated individually via SSM getParameters

## New Intended Behavior
1. Secrets are populated at the beginning of every job via `populate-env-vars-into-job` step
1. All values under SSM path of an environment are pulled and added to environment in one call via SSM:getParametersByPath 
2. As part of this effort: 
   - values in SSM have been added or moved to match key name with path (see Details section below)
   - these values will be removed from S3 configs once this change is populated to all services in each environment

Confirmed to work locally.
@codin-mgm I will need the mgm-ci-bot to have access to `GetParametersByPath` before this can be merged.

#### S3 Keys migrated to SSM:

  CLUSTER,
  ANDROID_PLATFORM_ARN,
  AWS_GATEWAY_URL,
  AWS_GW_API_KEY,
  AWS_REGION,
  COLLECTOR_AGENT_HOST,
  COLLECTOR_AGENT_PORT,
  DEVICE_REGISTRATION_TABLE,
  DEVICE_REGISTRATION_KEY,
  GSE_HOST,
  GSE_ORIGIN,
  IOS_PLATFORM_ARN,
  LOG_BUCKET,
  LOG_GROUP_PREFIX,
  MGMRESORTS_HOST,
  MGMRESORTS_PROPERTY_ID,
  NODE_ENV,
  OKTA_DOMAIN,
  OKTA_PREVIEW,
  SENTRY_DSN,
  TASK_ROLE_ARN,
  TIBCO_HOST,
  EXECUTION_ROLE_ARN,
  LOG_PUBLIC_IP,
  OKTA_CLIENT_ID,
  OKTA_CLIENT_SECRET,
  OKTA_AUTHORIZATION_SERVER_ID,
  LAMBDA_EXECUTION_ROLE,
  SUBNETS,
  SECURITY_GROUP,
  GSE_SSL_CERT,
  CREDENTIAL_SERVICES_HOST,
  LEGIC_HOST,
  TOKENIZATION_HOST,
  DMP_STATIC_HOST,
  SNT_AUTH_HOST,
  SNT_HOST,
  TIBCO_EAIWS_HOST,
  TIBCO_WS_HOST,

#### SSM keys moved
  SNT/ClientCredentials/client_id ==> SNT_CLIENT_ID
  SNT/ClientCredentials/client_secret ==> SNT_CLIENT_SECRET
  okta/token ==> OKTA_TOKEN
  Gateway/apikey ==> AWS_GW_API_KEY
  Cred_Svc/username ==> CRED_SVC_USERNAME
  Cred_Svc/password ==> CRED_SVC_PASSWORD
  Cred_Svc/Callback/username ==> CRED_SVC_CALLBACK_USERNAME
  Cred_Svc/Callback/password ==> CRED_SVC_CALLBACK_PASSWORD
  Dormakaba/wallet ==> WALLET_ID
  Legic/username ==> LEGIC_USERNAME
  Legic/password ==> LEGIC_PASSWORD

### CODE REVIEW

**Review by patrick-sf — DISMISSED** [review]

**Review by misiek-sf — DISMISSED** [review]

**Review by misiek-sf — DISMISSED** [review]

**Review by tomasz-sf — DISMISSED** [review]

**Comment by afaq-sf** [comment]
# PR Blocked by mgm-ci-bot access
Code is ready for review but complete testing / merging / publishing is blocked by `mgm-ci-bot` not having access to `SSM:GetParametersByPath` action.

Access has been requested from security team.

**Comment by afaq-sf** [comment]
@codin-mgm @misiek-sf @patrick-sf @tomasz-sf
We are not able to fetch SSM keys for `/MobApp/Development/*` paths due the same KMS key ID issue as production.

However, QA4, UAT, and Preprod seem to work for manual testing. For example, running `aws ssm get-parameters-by-path --path /MobApp/QA4/ --with-decryption` will work and fetch all values successfully.

This means that builds using SSM params will function in QA4, UAT, and Preprod 
but will fail for Development.

I would like to re-open this PR for review and move forward knowing that Development will fail until the KMS Key ID fix is made and the other envs can at least move forward for now.

**Review by codin-mgm — DISMISSED** [review]

**Review by misiek-sf — DISMISSED** [review]

**Review by codin-mgm — APPROVED** [review]

**Pull request merged**
